src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -442,4 +446,50 @@ private boolean checkAcceptedBody(String accept) {
         }
         return false;
     }
+
+    /**
+     * Build the request URI for the future consumer requests
+     *
+     * @param routingContext context of the current HTTP request
+     * @return the request URI for the future consumer requests
+     */
+    private String buildRequestUri(RoutingContext routingContext) {
+        String baseUri = routingContext.request().absoluteURI();
+
+        String xForwarded = routingContext.request().getHeader("x-forwarded");
+        if (xForwarded != null && !xForwarded.isEmpty()) {
+            Matcher hostMatcher = hostPattern.matcher(xForwarded);
+            Matcher protoMatcher = protoPattern.matcher(xForwarded);
+            if (hostMatcher.find() && !hostMatcher.group(1).isEmpty() &&
+                protoMatcher.find() && !protoMatcher.group(1).isEmpty()) {
+
+                baseUri = this.buildBaseUri(routingContext, hostMatcher.group(1), protoMatcher.group(1));
+            }
+        } else {
+            String xForwardedHost = routingContext.request().getHeader("x-forwarded-host");
+            String xForwardedProto = routingContext.request().getHeader("x-forwarded-proto");
+            if (xForwardedHost != null && !xForwardedHost.isEmpty() &&
+                xForwardedProto != null && !xForwardedProto.isEmpty()) {
+
+                baseUri = this.buildBaseUri(routingContext, xForwardedHost, xForwardedProto);
+            }
+        }
+        return baseUri;
+    }
+
+    private String buildBaseUri(RoutingContext routingContext, String host, String proto) {
+        String[] hostSplit = host.split(":");
```

Comment:
What if host is an ipv6 address, e.g. the url starts `http://[1080:0:0:0:8:800:200C:417A]:8080/`? 

I suppose `host.matches("^.*:[0-9]+$")` is probably sufficient, but you should probably factor out a constant `Pattern`.